packages/dev/core/src/Meshes/greasedLineMesh.ts

```diff
@@ -0,0 +1,535 @@
+            this._counters.push(...counters);
+            this._previous.push(...previous);
+            this._next.push(...next);
+            this._uvs.push(...uvs);
+            this._side.push(...side);
+        });
+
+        if (!this._lazy) {
+            this._drawLine();
+            this._updateRaycastBoundingInfo();
+        }
+    }
+
+    /**
+     * Clones the line.
+     * @param name new line name
+     * @param newParent new parent node
+     * @returns cloned line
+     */
+    public clone(name: string = `${this.name}-cloned`, newParent?: Nullable<Node>) {
+        // TODO: can this method be async and can we use dynamic imports of the material?
+
+        const lineOptions = {};
+        DeepCopier.DeepCopy(this._options, lineOptions);
+        const materialOptions = {};
+        DeepCopier.DeepCopy(this._pluginMaterial.getOptions(), materialOptions);
+
+        const material =
+            (<GreasedLineMaterialOptions>materialOptions).materialType === GreasedLineMeshMaterialType.MATERIAL_TYPE_PBR
+                ? new PBRMaterial(name, this._scene)
+                : new StandardMaterial(name, this._scene);
+        const pluginMaterial = new GreasedLinePluginMaterial(material, this._scene, materialOptions);
+        const cloned = new GreasedLineMesh(name, this._scene, <GreasedLineMeshOptions>lineOptions, pluginMaterial);
+        cloned.material = material;
+
+        if (newParent) {
+            cloned.parent = newParent;
+        }
+
+        return cloned;
+    }
+
+    /**
+     * Serializes this ground mesh
+     * @param serializationObject object to write serialization to
+     */
+    public serialize(serializationObject: any): void {
+        super.serialize(serializationObject);
+        serializationObject.lineOptions = this._options;
+        serializationObject.materialOptions = this._pluginMaterial.getOptions();
+    }
+
+    /**
+     * Parses a serialized ground mesh
+     * @param serializedMesh the serialized mesh
+     * @param scene the scene to create the ground mesh in
+     * @returns the created ground mesh
+     */
+    public static Parse(serializedMesh: any, scene: Scene): Mesh {
+        const lineOptions = <GreasedLineMeshOptions>serializedMesh.lineOptions;
+        const materialOptions = <GreasedLineMaterialOptions>serializedMesh.materialOptions;
+        const name = <string>serializedMesh.name;
+
+        const material =
+            materialOptions.materialType === GreasedLineMeshMaterialType.MATERIAL_TYPE_PBR ? new PBRMaterial(name, scene) : new StandardMaterial(name, scene);
+        const pluginMaterial = new GreasedLinePluginMaterial(material, scene, materialOptions);
+
+        const parsed = new GreasedLineMesh(name, scene, lineOptions, pluginMaterial);
+        parsed.material = material;
+        return parsed;
+    }
+
+    // TODO: which parameters to suppport?
+    public intersects(
+        ray: Ray,
+        fastCheck?: boolean | undefined,

```

Comment:
No need for the "| undefined" part, it is implicit when using "?" (same thing below).

Reply:
Copied it from existing BabylonJS code :-) . Fixed.